app/controllers/animals_controller.rb

```diff
@@ -0,0 +1,45 @@
+class AnimalsController < ApplicationController
+
```

Comment:
All methods are nicely written. Code is clean and very readable. Making it very easy to keep track and understand